New package: CyberSift.TutelaWindowsAgent version 1.5.0

dvas0004 opened the pull request:
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/156793)

dvas0004 committed:
New package: CyberSift.TutelaWindowsAgent version 1.5.0

stephengillie commented:
Automatic Validation ended with:

> Installation failed with exit code -1978335187
> 2024-06-06 08:42:54.635 [FAIL] Installer failed security check. Url: http://repo.cybersift.io/cs-vas/tutela.msi Result: 0x80004005

Automated error analysis suggests -1978335187 may mean Installer failed security check. Another script should retry this every 18 hours. <!-- [Policy] Needs-Attention`n[Policy] Validation-Defender-Error -->

(Automated response - build 883.)

Trenly commented on manifests/c/CyberSift/TutelaWindowsAgent/1.5.0/CyberSift.TutelaWindowsAgent.locale.en-US.yaml:
```diff
@@ -0,0 +1,12 @@
+Publisher: CyberSift
+PackageName: Tutela Windows Agent
+License: Commercial
+ShortDescription: Tutela Windows Agent
```
Can you please provide a better description of what this software is or what it is used for?

dvas0004 replied:
Yes sure!
Updated short description with: `Tutela Windows Agent. This agent enumerates software running on a workstation to check for applicable CVEs. It also runs several CIS benchamrk checks. (https://www.cybersift.com/tutela)`

dvas0004 committed:
Update CyberSift.TutelaWindowsAgent.locale.en-US.yaml

Providing additional details for https://github.com/microsoft/winget-pkgs/pull/156793

stephengillie commented:
Automatic Validation ended with:

> Installation failed with exit code -1978335226
> 2024-06-06 23:52:40.262 [CLI ] ShellExecute installer failed: 1603
> WixQuietExec:  Error 0x800700d8: Command failed to execute.
> WixQuietExec:  Error 0x800700d8: QuietExec Failed
> WixQuietExec:  Error 0x800700d8: Failed in ExecCommon method
> CustomAction CustomInstallExec0 returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
> Property(S): ErrorDialog = ErrorDlg
> MSI (s) (B0:68) [23:52:40:070]: Product: Tutela Windows Agent -- Installation failed.

Automated error analysis suggests -1978335226 may mean Install successful but system restart may be required.

Automated error analysis suggests 0x800700d8 may mean This version of this application is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher. (Should this have `Architecture: x64`?)

Automated error analysis suggests 1603 may mean Install successful but system restart may be required. Should this be add as an `InstallerSuccessCode` or similar?

(Automated response - build 883.)

stephengillie commented:
Manual Validation ended with:
> 
> 2024-06-07 08:30:38.815 [CLI ] Package hash verification failed. SHA256 in manifest [1d42750a0dd26fe9672f504d598a54caa692c70fed1a42d4e6ddce251302075c] does not match download [01efe25b1968ed68c160d256c7b1c55e7524978bf24c279c23ac3fab25ea8e2a]

(Automated response - build 883.)

dvas0004 committed:
Update CyberSift.TutelaWindowsAgent.installer.yaml

dvas0004 commented:
@Trenly @stephengillie  checks finally passing... thank you so much for your helpful advice!! 🙂

stephengillie commented:
Hi @dvas0004,

I'm happy to help. 😄 

Verify:
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/57dbf0da-9dca-4b3e-854f-fdbcb5502ab9)
- `updater.exe` has an error - is it supposed to be run from command line? 

![image](https://github.com/microsoft/winget-pkgs/assets/114952565/309074ab-1bea-4978-86d2-b0816820f53e)
- `ETWReader.exe` needs `Microsoft.VCRedist.2015+.x64` (or `x86`) - should it be added as a dependency here? 

![image](https://github.com/microsoft/winget-pkgs/assets/114952565/1ff621e1-5a7f-497e-806e-59a26241e858)

dvas0004 committed:
Update CyberSift.TutelaWindowsAgent.installer.yaml

updating dependencies

dvas0004 committed:
Update CyberSift.TutelaWindowsAgent.installer.yaml

updated installer with newer `updater.exe`

dvas0004 commented:
Submitting new manifest to address the following:

> updater.exe has an error - is it supposed to be run from command line?

It should be able to run yes:
![image](https://github.com/microsoft/winget-pkgs/assets/5849965/880d40c0-9281-4424-8214-3c8348d4e9b1)

From the error it seems there was some issue to connect to the windows service manager. I've updated `updater.exe`  to handle errors here more gracefully

> ETWReader.exe needs Microsoft.VCRedist.2015+.x64 (or x86) - should it be added as a dependency here?

Added, thank you!

stephengillie commented:
Thanks for updating the updater and the manifest. All seems to be working as expected: 
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/1fa9a554-450a-4d99-b74e-0bb4c5b53326)
